src/commands/up.rs

```diff
@@ -280,7 +388,7 @@ impl UpCommand {
         let exec_opts = if self.help {
             TriggerExecOpts::NoApp
         } else {
-            let from_registry = self.reference.is_some();
+            let from_registry = self.registry_source.is_some();
```

Comment:
@itowlson Because this isn't necessarily set for a registry reference (e.g. `-f ghcr.io/alice/foobar:v1`) the trigger flag `--from-registry` will not properly be set when inferring an OCI reference from `-f` as the `AppSource`.

I'd suggest changing `prepare_locked_app` to explicitly accept a `from_registry: bool` parameter that is `true` when calling from `prepare_app_from_oci` and `false` for `prepare_app_from_bindle`.